### PR TITLE
Generate a consistent map order for Python and JS

### PR DESCRIPTION
**Pull request by jparise**
Constant map values are stored internally as a map of pointers so their
iteration order is determined by pointer comparison. The written order of map
entries in generate code is therefore not deterministic and can result in
unexpected diffs across multiple thrift compiler runs.

This change is a coarse attempt to preserve the original key ordering (as
parsed from the original Thrift IDL file) for the Python and JS code
generators.

### CODE REVIEW

File: compiler/cpp/src/generate/t_js_generator.cc [hunk 1]
```diff
@@ -551,15 +551,16 @@ string t_js_generator::render_const_value(t_type* type, t_const_value* value) {
     indent_up();
 
     const map<t_const_value*, t_const_value*>& val = value->get_map();
-    map<t_const_value*, t_const_value*>::const_iterator v_iter;
-    for (v_iter = val.begin(); v_iter != val.end(); ++v_iter) {
-      if (v_iter != val.begin())
+    const vector<t_const_value*>& keys = value->get_map_keys();
+    vector<t_const_value*>::const_iterator k_iter;
+    for (k_iter = keys.begin(); k_iter != keys.end(); ++k_iter) {
+      if (k_iter != keys.begin())
```
Comment (lytefast) [inline compiler/cpp/src/generate/t_js_generator.cc]: couldn't you get away with not doing this if you use a do/while loop?

Comment (jparise) [inline compiler/cpp/src/generate/t_js_generator.cc]: I was trying to make a minimal amount of structural changes so our fork doesn't drift too far from the main code line.

Comment (lytefast) [inline compiler/cpp/src/generate/t_js_generator.cc]: Makes sense. Perf probably doesn't really matter either.

**Comment by lytefast** [comment]
LGTM. I would just make note that the order is determined by input order.

**Pull request merged**